ui/page/transaction/transactions_page.go

```diff
@@ -222,93 +219,135 @@ func (pg *TransactionsPage) loadTransactions(offset, pageSize int32) ([]*sharedW
-										txt.Color = pg.Theme.Color.GrayText3
-										gtx.Constraints.Min.X = gtx.Constraints.Max.X
-										return layout.Center.Layout(gtx, func(gtx C) D {
-											return layout.Inset{Top: padding, Bottom: padding}.Layout(gtx, txt.Layout)
+							return pg.scroll.List().Layout(gtx, 1, func(gtx C, i int) D {
+								return layout.Inset{Right: values.MarginPadding2}.Layout(gtx, func(gtx C) D {
+									return card.Layout(gtx, func(gtx C) D {
+										if itemCount == -1 || pg.showLoader {
+											gtx.Constraints.Min.X = gtx.Constraints.Max.X
+											return layout.Center.Layout(gtx, pg.materialLoader.Layout)
+										}
+
+										wallTxs := pg.scroll.FetchedData()
+										return pg.transactionList.Layout(gtx, len(wallTxs), func(gtx C, index int) D {
+											tx := wallTxs[index]
+											return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+												layout.Rigid(func(gtx C) D {
+													return components.LayoutTransactionRow(gtx, pg.Load, wal, tx, true)
+												}),
+												layout.Rigid(func(gtx C) D {
+													// No divider for last row
+													if index == len(wallTxs)-1 {
+														return layout.Dimensions{}
+													}
+
+													gtx.Constraints.Min.X = gtx.Constraints.Max.X
+													separator := pg.Theme.Separator()
+													return layout.E.Layout(gtx, func(gtx C) D {
+														// Show bottom divider for all rows except last
+														return layout.Inset{Left: values.MarginPadding56}.Layout(gtx, separator.Layout)
+													})
+												}),
+											)
 										})
-									}
-
-									wallTxs := pg.scroll.FetchedData()
-									return pg.transactionList.Layout(gtx, len(wallTxs), func(gtx C, index int) D {
-										tx := wallTxs[index]
-										return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
-											layout.Rigid(func(gtx C) D {
-												return components.LayoutTransactionRow(gtx, pg.Load, wal, tx, true)
-											}),
-											layout.Rigid(func(gtx C) D {
-												// No divider for last row
-												if index == len(wallTxs)-1 {
-													return layout.Dimensions{}
-												}
-
-												gtx.Constraints.Min.X = gtx.Constraints.Max.X
-												separator := pg.Theme.Separator()
-												return layout.E.Layout(gtx, func(gtx C) D {
-													// Show bottom divider for all rows except last
-													return layout.Inset{Left: values.MarginPadding56}.Layout(gtx, separator.Layout)
-												})
-											}),
-										)
 									})
 								})
 							})
 						})
 					})
 				}),
 				layout.Expanded(func(gtx C) D {
+					if pg.isHomepageLayout && pg.walletDropDown != nil {
+						return pg.walletDropDown.Layout(gtx, 0, false)
+					}
+					return D{}
+				}),
+				layout.Expanded(func(gtx C) D {
+					if pg.walletNotReady() && pg.isHomepageLayout {
+						return D{}
+					}
 					return pg.txTypeDropDown.Layout(gtx, 0, true)
 				}),
```

Comment:
Instead of these additional widgets in the stack that could be empty, only display useful widgets.
Something like this should work:

```
txListWidget :=  layout.Inset{Top: values.MarginPadding60}.Layout(gtx, func(gtx C) D {
	return pg.showDisabledLayout(gtx, func(gtx C) D {...}
})

if pg.walletNotReady() && pg.walletDropDown == nil {
	return txListWidget // nothing else to display on this page at this time
}

pageElements := []layout.StackChild{layout.Expanded(txListWidget)}
if pg.walletDropDown != nil { // display wallet dropdown if available
	walletDropdownWidget := layout.Expanded(func(gtx C) D {
		return pg.walletDropDown.Layout(gtx, 0, false)
	})
	pageElements = append(pageElements, walletDropdownWidget)
}
if !pg.walletNotReady() { // display tx dropdown if selected wallet is ready
	txDropdownWidget := layout.Expanded(func(gtx C) D {
		return pg.txTypeDropDown.Layout(gtx, 0, false)
	})
	pageElements = append(pageElements, txDropdownWidget)
}

return layout.Stack{Alignment: layout.NW}.Layout(gtx, pageElements...)
```